lib/blocking_queue.ex

```diff
@@ -100,18 +106,36 @@ defmodule BlockingQueue do
 
   `pid` is the process ID of the BlockingQueue server.
   `item` is the value to be pushed into the queue.  This can be anything.
+  `timeout` (optional) is the timeout value passed to GenServer.call (does not impact how long pop will wait for a message from the queue)
   """
-  @spec push(pid, any) :: nil
-  def push(pid, item), do: GenServer.call(pid, {:push, item})
+  @spec push(pid, any, integer) :: nil
+  def push(pid, item, timeout \\ 5000) do
+    case GenServer.call(pid, {:push, item}, timeout) do
+      :block ->
+        receive do
+          :awaken -> :ok
+        end
+      _ -> nil
+    end
+  end
```

Comment:
I know you had a comment about this, but this is basically exactly what I was expecting.